app/services/p2p_service.py

```
@@ -0,0 +1,539 @@
+            dest_path = os.path.join(local_path.source_path, filename)
+            if os.path.exists(dest_path):
+                failed.append({"id": cache_id, "error": f"{filename} already exists in the destination path"})
+                continue
+
+            encoded_id = quote(cache_entry.remote_file_id, safe="")
+            url = f"http://{peer.host}:{peer.port}/api/v1/p2p/files/{encoded_id}/content"
+            headers = {"X-FF-PSK": psk_hash}
+
+            try:
+                with httpx.Client(timeout=300.0) as client:
+                    with client.stream("GET", url, headers=headers) as response:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)